src/hooks/use-current-agent.js

```diff
@@ -24,7 +24,7 @@ import {
  * This is an example of switching dynamically between agents based on the Current Agent. TODO: some kind of registration mechanism.
  */
 
-const useCurrentAgent = ( { chat, toolkit } ) => {
+const useCurrentAgent = ( { chat, toolkit, newAgent } ) => {
```

Comment:
This doesn't belong here. Please create a new hook called something like `useAgent( agentclassname )` rather than bolting this onto the side of `useCurrentAgent()` :)

Reply:
Also we should probably not be passing around classes or class instances in the Redux context. It's asking for trouble. Agents should be registered with the system and their instances created by some kind of agent manager which can create them based on a unique string ID.

Reply:
For sure, this was never intended for actual implementation, rather just to get my head around how things were currently setup so we could see what was actually needed in terms of a proper API for external use. Will just wait for you to flesh out the ideas of a registry.